pkg/calculator/handler.go

```diff
@@ -69,22 +60,27 @@ func (ec *EmissionCalculator) Apply(event bus.Event) {
 		minWatts:      specs.MinWatts,
 		maxWatts:      specs.MaxWatts,
 		totalEmbodied: specs.TotalEmbodiedKiloWattCO2e,
-		cores:         mCPU.UnitAmount(),
-		usageCPU:      mCPU.Usage(),
 		pue:           emFactors.AveragePUE,
 		gridCO2e:      gridCO2e,
 	}
 
-	mCPU.SetEmissions(
-		v1.NewResourceEmission(
-			c.operationalCPUEmissions(cfg.Interval),
-			v1.GCO2eqkWh,
-		),
-	)
+	// calculate and set the operational emissions for each
+	// metric type (CPU, Memory, Storage, and networking)
+	metrics := instance.Metrics()
+	for _, v := range metrics {
+		em, err := c.operationalEmissions(&v, interval)
+		if err != nil {
+			klog.Errorf("error calculating operational emissions: %+v", err)
+			return
```

Comment:
dont return in a loop, you'll skip metrics that might be valid, maybe print an error and continue

Reply:
That's a good idea. Maybe I'lll just "collect" errrors and return a slice of them.